src/main/java/de/tum/in/www1/artemis/config/migration/setups/localvc/jenkins/MigrationEntryJenkinsToLocalVC.java

```diff
@@ -0,0 +1,141 @@
+package de.tum.in.www1.artemis.config.migration.setups.localvc.jenkins;
+
+import java.util.List;
+
+import jakarta.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.config.migration.setups.localvc.LocalVCMigrationEntry;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.SolutionProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.participation.TemplateProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.exception.JenkinsException;
+import de.tum.in.www1.artemis.repository.AuxiliaryRepositoryRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseStudentParticipationRepository;
+import de.tum.in.www1.artemis.repository.SolutionProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.TemplateProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.service.UriService;
+import de.tum.in.www1.artemis.service.connectors.jenkins.build_plan.JenkinsBuildPlanService;
+
+/**
+ * Migration of Jenkins Build plans when the repositories will move to LocalVC.
+ */
+@Component
+@Profile("jenkins")
```

Comment:
Only load when the migration profile is active?

Reply:
:heavy_check_mark: with 226fcdd9c8c46120325771ada92a09e35a8d7892